packages/paste-core/components/skeleton-loader/src/index.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useSpring, animated} from '@twilio-paste/animation-library';
+import {css, styled} from '@twilio-paste/styling-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {LayoutProps, MarginProps, BorderRadiusProps} from '@twilio-paste/style-props';
+
+const AnimatedSkeleton = animated(Box);
+
+const StyledAnimatedSkeleton = styled(AnimatedSkeleton)(() =>
+  css({
+    background: `linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.3) 40%, rgba(255, 255, 255, 0.3) 60%, transparent)`,
+  })
+);
+
+export interface SkeletonLoaderProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    LayoutProps,
+    MarginProps,
+    BorderRadiusProps {
+  disabled?: boolean;
+}
+
+const SkeletonLoader = React.forwardRef<HTMLDivElement, SkeletonLoaderProps>(
+  (
+    {
+      borderBottomLeftRadius,
+      borderBottomRightRadius,
+      borderRadius = 'borderRadius20',
+      borderTopLeftRadius,
+      borderTopRightRadius,
+      display,
+      disabled,
+      height = 'sizeIcon20',
+      margin,
+      marginBottom,
+      marginLeft,
+      marginRight,
+      marginTop,
+      marginX = null,
+      marginY = null,
+      maxHeight,
+      maxWidth,
+      minHeight,
+      minWidth,
+      size = null,
+      width = null,
+      ...props
+    },
+    ref
+  ) => {
+    const animatedSkeletonStyles = useSpring({
+      loop: true,
+      cancel: disabled,
+      from: {translateX: '-100%', skew: '155deg'},
+      to: {translateX: '100%', skew: '155deg'},
+      config: {
+        mass: 0.1,
+        tension: 80,
+        friction: 50,
+      },
+    });
+
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
```

Comment:
Would it be simpler if we didn't safelySpread here and just passed all the props we weren't setting values for? I'm not sure what the concern is here

Reply:
If the intention is only to allow Layout, Margin, BorderRadius props and HTML attributes as the possible API, then safely spreading props makes sense to me.

It feels awkward but I'm not sure we have a good solution for "I want these, but not those style props, plus all HTML attributes". We only have a solution for "I don't want any style props, but all HTML attributes"

Reply:
I more meant that we probably don't have any restrictions on style props for this other than the ones we're setting ourselves, which would override the spread. So like:
```
<Box {...props} backgroundColor="w/e">
```
Would allow passing any styles Box accepts, but makes sure our styles override.

Reply:
This has changed a bit to only pass down Layout and BorderRadius props. Consumers can use our other layout components for any necessary margins. I feel like scoping this down to just those is best for this first iteration because that adds scope to this component, and removes potentially unnecessary complexity. We could always add in complexity in the future if we get that feedback.